### DISCUSSION TITLE
InstrumentedAttribute property type of relationships changed in 2.0.29

### DISCUSSION THREAD

**Discussion by lunalucadou** [sqlalchemy/sqlalchemy#11350] [Usage Questions] [answered]
In 2.0.28, calling `InstrumentedAttribute.property` on a reference to another class returned an instance of `sqlalchemy.orm.Relationship`.

This makes sense, as per https://docs.sqlalchemy.org/en/20/orm/internals.html#sqlalchemy.orm.InstrumentedAttribute, the `.property` attribute is a `MapperProperty` which, when it is a reference to another class, should be an instance of `sqlalchemy.orm.Relationship`.

In 2.0.29, it returns an instance of `sqlalchemy.orm.relationships._RelationshipDeclared`, which makes a check in our codebase, `type(relationship_member.property) is Relationship`, return false.

I don't see anything in the release notes about this. Was this change intentional? If so, should I update the check to `type(relationship_member.property) is _RelationshipDeclared`? (Alternatively, is there a better way of checking if a property is a relationship?)

**Comment by CaselIT** [comment, marked as answer]
Hi,

That class was introduced to fix the bug reported in https://github.com/sqlalchemy/sqlalchemy/issues/10611

The check should probably use `isinstance` instead of using is on the `is`

**Reply by lunalucadou** [reply]
Thanks, I'll use `isintance(relationship_member.property, Relationship)` from here on out. I figured that item in the changelog had to be related to this but couldn't figure out how exactly.
